core/Pos/Core/Arbitrary.hs

```diff
@@ -62,6 +62,8 @@ instance Arbitrary Types.Address where
         Types.UnknownAddressType <$> choose (3, 255) <*> scale (min 150) arbitrary
         ]
 
+-- NOTE: Deriving an 'Arbitrary' instance for wrappers over integral types does not use
```

Comment:
I don't think this has to do with the type being an instance of `Integral`, as your comment may suggest. What's happening is that `deriving` for newtypes uses the `GND` deriving strategy, so the `shrink` method is inherited from the underlying type. Since `ChainDifficulty` is a newtype over `Word64`, we get a `shrink` impl that tries to bring the value to `0` (which is not strictly a "better" instance, depending on what values you consider to be trivial in your domain. For example, if you represent unique identifiers as numbers, you'd probably want no shrinking to take place).